Update to 3.14 and add gradle files

mandy-chessell opened the pull request:
Signed-off-by: Mandy Chessell <mandy.e.chessell@gmail.com>

This PR updates the code to run with 3.14.  It also adds gradle files - however the gradle build does not currently work - seems to be a problem with buildign the jars.

planetf1 reviewed (dismissed):
Additionally, the build throws some dependency issues which need resolving:
```
cat /Users/jonesn/IdeaProjects/egeria-dev-projects/build/reports/dependency-analysis/build-health-report.txt
Advice for :asset-set-up
Unused dependencies which should be removed:
  implementation 'org.odpi.egeria:platform-services-client:3.15-SNAPSHOT'

Advice for :component-id-report
Unused dependencies which should be removed:
  implementation 'org.odpi.egeria:cohort-registry-file-store-connector:3.15-SNAPSHOT'
  implementation 'org.odpi.egeria:inmemory-repository-connector:3.15-SNAPSHOT'
  implementation 'org.odpi.egeria:omrs-rest-repository-connector:3.15-SNAPSHOT'
  implementation 'org.odpi.egeria:open-metadata-archive-directory-connector:3.15-SNAPSHOT'
  implementation 'org.odpi.egeria:open-metadata-archive-file-connector:3.15-SNAPSHOT'
  implementation 'org.odpi.egeria:open-metadata-security-samples:3.15-SNAPSHOT'

Advice for :egeria-config-utility
Transitively used dependencies that should be declared directly as indicated:
  implementation 'org.odpi.egeria:ffdc-services:3.15-SNAPSHOT'
```

See the suggestion on data-report/build.gradle -- any modules built within this project should be referenced as files (ie internal to the build) rather than external dependencies.

planetf1 commented on gradle/wrapper/gradle-wrapper.properties:
```diff
@@ -0,0 +1,5 @@
+distributionBase=GRADLE_USER_HOME
+distributionPath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.1.1-bin.zip
```
The egeria-database-connectors is probably a better example that has adopted some of our recent learnings for gradle project builds. (Happy to review the gradle build once the basics are in place)

mandy-chessell replied:
I have copied the wrapper and properties from https://github.com/odpi/egeria-connector-jdbc

mandy-chessell commented on database-report/build.gradle:
```diff
@@ -0,0 +1,30 @@
+}
+
+dependencies {
+    implementation "org.odpi.egeria:egeria-report-utilities"
```
fixed

mandy-chessell commented:
> Additionally, the build throws some dependency issues which need resolving:
> 
> ```
> cat /Users/jonesn/IdeaProjects/egeria-dev-projects/build/reports/dependency-analysis/build-health-report.txt
> Advice for :asset-set-up
> Unused dependencies which should be removed:
>   implementation 'org.odpi.egeria:platform-services-client:3.15-SNAPSHOT'
> 
> Advice for :component-id-report
> Unused dependencies which should be removed:
>   implementation 'org.odpi.egeria:cohort-registry-file-store-connector:3.15-SNAPSHOT'
>   implementation 'org.odpi.egeria:inmemory-repository-connector:3.15-SNAPSHOT'
>   implementation 'org.odpi.egeria:omrs-rest-repository-connector:3.15-SNAPSHOT'
>   implementation 'org.odpi.egeria:open-metadata-archive-directory-connector:3.15-SNAPSHOT'
>   implementation 'org.odpi.egeria:open-metadata-archive-file-connector:3.15-SNAPSHOT'
>   implementation 'org.odpi.egeria:open-metadata-security-samples:3.15-SNAPSHOT'
> 
> Advice for :egeria-config-utility
> Transitively used dependencies that should be declared directly as indicated:
>   implementation 'org.odpi.egeria:ffdc-services:3.15-SNAPSHOT'
> ```
> 
> See the suggestion on data-report/build.gradle -- any modules built within this project should be referenced as files (ie internal to the build) rather than external dependencies.

The report is useful - 
* I have followed the suggestions for :egeria-config-utility.
* For asset-set-up I have implemented that part of the utility that uses the platform services client
* For component-id-report, the dependencies are needed at runtime since their classes are loaded when the report runs.  I have changed them to `runtimeOnly` so hopefully they are pulled into the jar.
